tests/upgrade/test_crossgrade.sh

```diff
@@ -365,19 +370,19 @@ resetCluster() {
     echo_and_run_or_die kubectl label namespace "${TEST_NAMESPACE}" istio-injection=enabled
 }
 
-# Returns 0 if the passed string has form "Code 200 : 6601 (94.6 %)" and the percentage is smaller than ${MIN_200_PCT_FOR_PASS}
-percent200sAbove() {
-    local s=$1
-    local regex="Code 200 : [0-9]+ \\(([0-9]+)\\.[0-9]+ %\\)"
-    if [[ $s =~ $regex ]]; then
-        local pct200s="${BASH_REMATCH[1]}"
-        if (( pct200s > MIN_200_PCT_FOR_PASS )); then
-            return 0
-        fi
-        return 1
-    fi
-    echo "No Code 200 percentage found in log."
-    return 1
+# Return 1 if the specific error code percentage exceed corresponding threshold
+errorPercentBelow() {
+     local s
+     s=$(grep "${2}" "${1}")
+     local regex="${2} : [0-9]+ \\(([0-9]+)\\.[0-9]+ %\\)"
```

Comment:
Maybe just hardcode the term `Code` again in the regex and the caller just provides the code (e.g. 503) he/she is interested it. In the end it is an implementation detail, right?

Reply:
Done